Backpack-SwiftUI/TextField/Classes/BPKTextField.swift

```diff
@@ -76,6 +86,22 @@ public struct BPKTextField: View {
         }
     }
     
+    private var prefixView: some View {
+        HStack {
+            switch prefixState {
+            case .text(let prefixText):
+                BPKText(prefixText, style: .bodyDefault)
+                    .foregroundColor(.textSecondaryColor)
+            case .searchIcon:
+                BPKIconView(BPKIcon(name: "search"))
+                    .foregroundColor(.textPrimaryColor)
+                    .accessibilityHidden(true)
+            case nil:
+                EmptyView()
```

Comment:
It's recommended to avoid using `EmptyView()`, please use `@ViewModifier` instead! Also you don't need an HStack here, you can do like so:

```swift
@ViewBuilder
private var prefixView: some View {
  if let prefix {
    switch prefix {
      case .text(let prefixText):
        BPKText(prefixText, style: .bodyDefault)
          .foregroundColor(.textSecondaryColor)
      case .searchIcon:
        BPKIconView(BPKIcon(name: "search"))
          .foregroundColor(.textPrimaryColor)
            .accessibilityHidden(true)
        }
    }
}

Reply:
Not using EmptyView() anymore, it was in a draft version. Agree about HStack, but the problem is in opaque types (because of the compiler can't know which type will be using in runtime here). However, I remembered about @ViewBuilder attribute which should help here with creating "conditional tree" of views. Thanks for noticing that.